core/domain/user_services.py

```diff
@@ -1412,8 +1414,24 @@ def update_profile_picture_data_url(
         profile_picture_data_url: str. New profile picture url to be set.
     """
     user_settings = get_user_settings(user_id, strict=True)
-    user_settings.profile_picture_data_url = profile_picture_data_url
-    save_user_settings(user_settings)
+    username = user_settings.username
+    if username is None:
+        raise utils.ValidationError(
+            'User should have a non-None username.')
```

Comment:
Why is this needed?

Reply:
Yeah, removed the check now we only check if the username is None or not, if it is then we return from there. Please note that this function gets called when a new user sign up and as the mobile-user can have None username we are directly returning from the function.